### ISSUE TITLE
Ability to use crypto hashes?

### ISSUE THREAD

**Issue by mgcrea** [digitalbazaar/forge#344]
I'm writing code that will only run on NodeJS, is there a way to leverage the crypto module to get better performance? Does it makes sense?

Expected this `cert.sign(privateKey, crypto.createHash('sha256'));` to work, but it looks like it is not compatible with `md.sha256.create()`.

**Comment by dlongley** [comment]
@mgcrea, yes it's possible, but you'll have to wrap the node API with a forge API. The forge API is close to, but not identical to the node one, but it should be fairly easy to wrap.

In future versions of forge we'll likely make use of native code automatic for hashes when available (unless explicitly disabled via flag).

**Comment by coolaj86** [comment]
Just in case this helps someone else:

```javascript
var sha256Buf = crypto.createHash('sha256').update("Hello, World!").digest();
var md = forge.util.createBuffer(sha256Buf.toString('binary'), 'binary');
var sig = privateKey.sign(md);
var sigBuf = new Buffer(forge.util.bytesToHex(sigF), "hex");
var sig64 = sigBuf.toString('base64');
```
